test/generic/Matrix-test.jl

```diff
@@ -403,6 +403,12 @@ end
 
    @test issquare(A)
 
+   @test A[1:end, 1:end] == A
```

Comment:
Maybe add a test using `begin`, like `if VERSION >= v"1.4"; A[begin:end, :] == A; end` (`begin` is supported at least since Julia 1.4).